### PR TITLE
Support pytest > 8.0:

### PR DESCRIPTION
**Pull request by petrus-v**
Since pytest v8 we are receiving more file in _find_manifest_path method that method should not return a non present file in case __manifest__ is not found in 5 parents directory

### CODE REVIEW

**Comment by petrus-v** [comment]
This seems to work when using hatch-odoo when odoo addons are properly defined in `odoo/addons` namespace sub directory (test using odoo 14.0 and 17.0).

Probably #66 is better to overwrite module names while pytest is importing python modules (not test yet).

**Comment by petrus-v** [comment]
I'll explore an other way to hack the `odoo.addons` namespace, monkey patching the `odoo.models.MetaModel`...

**Comment by petrus-v** [comment]
Those two later commits make it works in many environment cases and make code even shorter than #66 @simahawk @micdahl do you mind to test this PR in your context ?

I've tested different scenario:
* with hatch-odoo using or not odoo/addons sub directory on version 14.0 and 17.0
* without hatch-odoo on version 14.0 (with pytest 7.4.4 and pytest 8.3.2)

`odoo.models.MetaModel` exists from Odoo version 10.0 so it might works from there ?!

**Comment by petrus-v** [comment]
For record, I got some issues testing api endpoints using [extendable](https://pypi.org/project/extendable/) classes

```
    def __getitem__(self, key: str) -> main.ExtendableMeta:
>       return self._extendable_classes[key]
E       KeyError: 'my_odoo_module.pydantic_models.ondemand.OnDemandOrder'
```
But using absolute import instead relative import in my test case fix the matter

**Comment by micdahl** [comment]
Hey @petrus-v , thanks for the commit. I would appreciate if there was support for pytest 8 in pytest-odoo.
I have tested your commit with an Odoo 16.0 Module and worked fine.

**Comment by petrus-v** [comment]
hum... maybe I should handle all kind of classes created in such odoo module not only those using the MetaModel Metaclass :thinking: 

ie: if we have following file `my_module/models/my_odoo_model.py`:

```python
class MyException(Exception):
  pass

class MyOdooModel(models.Model):
  _name = "my.odoo.model"

  @api.model
  def some_action(self):
    raise MyException("tada")
```

And want to test the raising exception, depending the import the commented one won't works

```python
# this absolute import won't works as MyException do not use the MetaModel Metaclass
# so the class namespace won't be set as expect
# from odoo.addons.my_module.models.my_odoo_models import MyException

# instead we have to import it using relative import
from ..models.my_odoo_models import MyException

from odoo.tests import TransactionCase

class MyTestCase(TransactionCase):
  def test_some_action(self):
     with self.assertRaises(MyException):
        self.env["my.odoo.model"].some_action()
```

File: pytest_odoo.py [hunk 1]
```diff
@@ -20,6 +20,16 @@
 import pytest
 from _pytest._code.code import ExceptionInfo
 
+def monkey_patch_odoo_meta_model():
+    originalMetaModel_new = odoo.models.MetaModel.__new__
```
Comment (simahawk) [inline pytest_odoo.py:24]: an explanation on why this is needed would be nice

File: pytest_odoo.py [hunk 2]
```diff
@@ -131,103 +141,16 @@ def enable_odoo_test_flag():
     odoo.tools.config['test_enable'] = False
 
 
-# Original code of xmo-odoo:
```
Comment (simahawk) [inline pytest_odoo.py:134]: no need for this ugly hack anymore? :exploding_head:

File: tests/test_pytest_odoo.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+from pytest_odoo import _find_manifest_path
```
Comment (simahawk) [inline tests/test_pytest_odoo.py:1]: Note that I switched to GH actions -> can you rebase please?
Next: introduce pre-commit :)

**Comment by simahawk** [comment]
> ```python
> # this absolute import won't works as MyException do not use the MetaModel Metaclass
> # so the class namespace won't be set as expect
> # from odoo.addons.my_module.models.my_odoo_models import MyException
> ```

hmmm this is not good I think

**Comment by petrus-v** [comment]
> > ```python
> > # this absolute import won't works as MyException do not use the MetaModel Metaclass
> > # so the class namespace won't be set as expect
> > # from odoo.addons.my_module.models.my_odoo_models import MyException
> > ```
> 
> hmmm this is not good I think

Yes I agree ! I'm closing it

File: pytest_odoo.py (hunk 2 above)
Comment (petrus-v) [reply pytest_odoo.py:134]: no this PR do not needs it but as mentioned this PR do not support python class created and used in a same python mobule while he does'nt respect python namespace and rely on Odoo addons_path.

In fact the metaclass monkey patch present on this PR was used to this purpose to hijack class namespace while importing model class. That's why it doesn't works with normal python class !

Unfortunately (and hopefully) we can't monkey patch `type` built-in :sweat_smile:

File: tests/test_pytest_odoo.py (hunk 3 above)
Comment (petrus-v) [reply tests/test_pytest_odoo.py:1]: good news

**Pull request closed without merging**